src/main/java/org/betonquest/betonquest/objectives/BlockObjective.java

```diff
@@ -28,42 +30,64 @@ public class BlockObjective extends CountingObjective implements Listener {
 
     private final boolean noSafety;
 
+    private final CompoundLocation location;
+
+    private final boolean hasLocation;
+
+    private final boolean ignorecancel;
+
     public BlockObjective(final Instruction instruction) throws InstructionParseException {
         super(instruction);
         selector = instruction.getBlockSelector();
         exactMatch = instruction.hasArgument("exactMatch");
         targetAmount = instruction.getVarNum();
         noSafety = instruction.hasArgument("noSafety");
+        final String stringlocation = instruction.getOptional("loc", "false");
+        location = "false".equalsIgnoreCase(stringlocation) ? null : new CompoundLocation(instruction.getPackage(), stringlocation);
+        hasLocation = !"false".equalsIgnoreCase(stringlocation);
+        ignorecancel = instruction.hasArgument("ignorecancel");
+
     }
 
     @Override
     public String getDefaultDataInstruction(final Profile profile) {
         return String.valueOf(targetAmount.getInt(profile));
     }
 
-    @EventHandler(ignoreCancelled = true, priority = EventPriority.HIGHEST)
-    public void onBlockPlace(final BlockPlaceEvent event) {
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onBlockPlace(final BlockPlaceEvent event) throws QuestRuntimeException {
```

Comment:
The exception should probably be handled instead of being passed on to the Bukkit event handler.

Reply:
Exception has been handled now.